usr/share/rear/rescue/GNU/Linux/290_kernel_cmdline.sh

```diff
@@ -1,51 +1,60 @@
-# purpose of the script is to detect some important KERNEL CMDLINE options on the current system
-# we should also use in rescue mode (automatically update KERNEL_CMDLINE array variable).
+# Purpose of the script is to get the COPY_KERNEL_PARAMETERS on the current system
+# to be used in the rescue/recovery system via automated update of KERNEL_CMDLINE.
 
-# Scanning current kernel cmdline to look for important option ($COPY_KERNEL_PARAMETERS) to include in KERNEL_CMDLINE
-for current_kernel_option in $( cat /proc/cmdline ); do
+# Also protect the rescue/recovery system by removing net.ifnames=0 from KERNEL_CMDLINE
+# if net.ifnames=0 is in KERNEL_CMDLINE but persistent network interface naming is used:
+local persistent_naming='no'
+is_persistent_ethernet_name $( ip route | awk '$2 == "dev" && $8 == "src" { print $3 }' | sort -u | head -n1 ) && persistent_naming='yes'
+
+# Scan current kernel cmdline for options in COPY_KERNEL_PARAMETERS to be included in KERNEL_CMDLINE:
+local current_kernel_option
+local new_kernel_options_to_add=()
+for current_kernel_option in $( cat /proc/cmdline ) ; do
     # Get the current kernel option name (part before leftmost "=") and
     # add the whole option (with value) to new_kernel_options_to_add array
     # if the option name is part of COPY_KERNEL_PARAMETERS array:
-    if IsInArray "${current_kernel_option%%=*}" "${COPY_KERNEL_PARAMETERS[@]}" ; then
-        new_kernel_options_to_add+=( "$current_kernel_option" )
-    fi
+    IsInArray "${current_kernel_option%%=*}" "${COPY_KERNEL_PARAMETERS[@]}" && new_kernel_options_to_add+=( "$current_kernel_option" )
```

Comment:
Cosmetic/stylistic remark : I actually think that replacing structured `if` `then` `fi` blocks by one-liners using shortcut evaluation decreases readability. But this is of course a very subjective topic with a strong bikeshed potential...

Reply:
I agree:
https://github.com/rear/rear/pull/2844/commits/4b4efc580a5ac7c4b8670b2ca4e511ccceec03bd

I was unsure if my one line 'CONDITION && COMMAND' is good here.
I used it because in general I prefer code with less lines
so I can see more code lines on a screen or in a window
to get a better overview of longer code parts.